src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -35,44 +37,89 @@ def get_latest_dataset_with_survey_id_and_period_id(
 
         return dataset_metadata
 
-    def perform_new_dataset_transaction(
+    def perform_batched_dataset_write(
```

Comment:
this function name is used in the writer service and firebase repository. could we rename one to avoid any confusion.
(perform_batched_dataset_write is called within perform_batched_dataset_write)

Reply:
Will change the name